presto-main/src/main/java/com/facebook/presto/operator/TableFinishOperator.java

```diff
@@ -156,12 +243,45 @@ public Page getOutput()
         return page.build();
     }
 
+    private List<ComputedStatistics> getComputedStatistics()
+    {
+        ImmutableList.Builder<ComputedStatistics> statistics = ImmutableList.builder();
+        while (!statisticsAggregation.isFinished()) {
+            Page page = statisticsAggregation.getOutput();
+            if (page == null) {
+                continue;
```

Comment:
you should yield here (return). Operator shouldn't do a ton of work within single call, otherwise a query might be "unkillable"

Reply:
IRL we cannot get here more data than a single page. We group the statistics on `per-partition` basis. And we never insert more than 100 partitions at once by default. But even if we inserted 100_000 - there still will be not enough data for more than a several pages. But than you would definitely have more severe problems. So for the sake of code simplicity in this class (which is already complex), i would go with what we have now.

Reply:
fine, but please add a comment that we deliberately not yield here